core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -939,7 +967,12 @@ private static Cache<String, AuthSession> newSessionCache(Map<String, String> pr
     return Caffeine.newBuilder()
         .expireAfterAccess(Duration.ofMillis(expirationIntervalMs))
         .removalListener(
-            (RemovalListener<String, AuthSession>) (id, auth, cause) -> auth.stopRefreshing())
+            (RemovalListener<String, AuthSession>)
+                (id, auth, cause) -> {
+                  if (null != auth) {
```

Comment:
Why did this need to change? The value isn't a soft reference so it shouldn't ever be null.

Reply:
it wasn't strictly necessary, but we've been doing null checks in all other instances of such a cache. I've reverted that change